peppy/project.py

```diff
@@ -594,56 +594,42 @@ def _auto_merge_duplicated_names(self):
             specified in the config
         """
         sample_names_list = [getattr(s, self.st_index) for s in self.samples]
-        dups_set = set(
-            [
-                x
-                for x in track(
-                    sample_names_list,
-                    description="Detecting duplicate sample names",
-                    disable=not self.is_sample_table_large,
-                )
-                if sample_names_list.count(x) > 1
-            ]
-        )
-        if not dups_set:
-            # all sample names are unique
+        duplicated_sample_ids = self._get_duplicated_sample_ids(sample_names_list)
+
+        if not duplicated_sample_ids:
```

Comment:
Maybe, `if len(duplicated_sample_ids) == 0:` ?

Small thing, just slightly better readability.

Reply:
According to PEP8 
![image](https://user-images.githubusercontent.com/43926522/185230216-cf2a12d0-a7ed-4354-b468-8729a4184b42.png)

Reply:
https://peps.python.org/pep-0008/